### PR TITLE
DefaultPutResolver allow overriding id field name

### LINKED ISSUE #165: DefaultPutResolver should allow overriding "id" field name

### PR DESCRIPTION
**Pull request by nikitin-da**
Closes #165 
@artem-zinnatullin Did you mean something like this?

### CODE REVIEW

File: storio-sqlitedb/src/androidTest/java/com/pushtorefresh/storio/sqlitedb/impl/InsertTest.java [hunk 1]
```diff
@@ -70,4 +73,36 @@
...
+        final PutResult putResult = storIOSQLiteDb
+                .put()
+                .object(tweet)
+                .withMapFunc(Tweet.MAP_TO_CONTENT_VALUES)
+                .withPutResolver(Tweet.PUT_RESOLVER)
+                .prepare()
+                .executeAsBlocking();
+
+        assertNotNull(putResult);
+        assertEquals(true, putResult.wasInserted());
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/androidTest/java/com/pushtorefresh/storio/sqlitedb/impl/InsertTest.java]: `assertTrue(putResult.wasInserted())`

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/DefaultPutResolver.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ private PutResult updateOrInsert(@NonNull StorIOSQLiteDb storIOSQLiteDb, @NonNul
         final int numberOfUpdatedRows = storIOSQLiteDb.internal().update(
                 new UpdateQuery.Builder()
                         .table(table)
-                        .where(BaseColumns._ID + "=?")
+                        .where(getIdFieldName() + "=?")
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/DefaultPutResolver.java]: What about to pass `idFieldName` to `updateOrInsert()` as parameter for better performance?

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/DefaultPutResolver.java [hunk 3]
```diff
@@ -75,4 +75,9 @@ private PutResult updateOrInsert(@NonNull StorIOSQLiteDb storIOSQLiteDb, @NonNul
                 ? PutResult.newUpdateResult(numberOfUpdatedRows, table)
                 : insert(storIOSQLiteDb, contentValues);
     }
+
+    @NonNull
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/DefaultPutResolver.java]: Missing Javadoc

**Comment by artem-zinnatullin** [comment]
Yes, exactly! 

But I thought unit test will be enough :) Unit test definitely required

**Comment by artem-zinnatullin** [comment]
@nikitin-da I'm writing `PutResolver` for `StorIOContentProvider` and I just realised that "id" can be `Object`, it can be `Long`, `String`, whatever thing convertable to `String`, because `SQLite` operates with queries arguments as `Strings` 

You can fix your PR and I'll change `id` to `Object` later, or you can do it as part of this PR, I'll create an issue for that

**Comment by nikitin-da** [comment]
Roger that

**Comment by nikitin-da** [comment]
Should I save integration test or it will be excess?

**Comment by artem-zinnatullin** [comment]
Save them, I think they will be useful :)

**Comment by nikitin-da** [comment]
Updated=)

**Comment by artem-zinnatullin** [comment]
Great!

**Pull request merged**